### ISSUE TITLE
Gradle setup instructions are not up to date

### ISSUE THREAD

**Issue by angryziber** [JetBrains/spek#142]
Gradle setup instructions are not up to date.

Junit platform now requires a `filters` to choose an engine:

```
junitPlatform {
  filters {
    engines {
      include 'spek'
    }
  }
}
```

And actually, it works for me also without these lines at all - remove them and it still finds the spek engine.

However, the whole Junit5 requirement is currently a disaster, as it is not stable yet.
IDEA Spek plugin doesn't work properly - it is missing a lot of features from Junit4 runner (Ctrl+Shift+T, Ctrl+Shift+F10, and other shortcuts). And now, after some updates from Junit5 (does spek depend on any latest version of it?), I cannot run tests from IDEA at all...

```
Dec 13, 2016 11:04:24 PM org.junit.platform.launcher.core.ServiceLoaderTestEngineRegistry loadTestEngines
INFO: Discovered TestEngines with IDs: [spek]
Exception in thread "main" java.lang.NoClassDefFoundError: org/junit/platform/engine/discovery/ClasspathRootSelector
	at org.jetbrains.spek.engine.SpekTestEngine.resolveSpecs(SpekTestEngine.kt:53)
	at org.jetbrains.spek.engine.SpekTestEngine.discover(SpekTestEngine.kt:37)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:108)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:84)
	at org.jetbrains.spek.tooling.runner.junit.JUnitPlatformSpekRunner.run(JUnitPlatformSpekRunner.kt:86)
	at org.jetbrains.spek.tooling.MainKt.main(Main.kt:58)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:147)
Caused by: java.lang.ClassNotFoundException: org.junit.platform.engine.discovery.ClasspathRootSelector
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 11 more
```

Which was the last version of Spek that run with Junit4 by default (I don't want to annotate all the classes)? It seems that at least 1.0.25 did work properly...

You can review the situation by cloning this:
https://github.com/angryziber/picasa-gallery/tree/spek-junit5

**Comment by angryziber** [comment]
I have reported the plugin bug there:
https://github.com/raniejade/spek-idea-plugin/issues/24

However, I would like to use IDEA's built-in Junit5 support instead...

**Comment by raniejade** [comment]
You have to add the proper dependency to `junit-platform-engine`.

```gradle
dependencies {
    ...
    testRuntime 'org.junit.platform:junit-platform-launcher:1.0.0-M3'
    testRuntime 'org.jetbrains.spek:spek-junit-platform-engine:1.1.0-beta2'
}
```

The JUnit 5 plugin for IDEA is for the JUnit Jupiter Engine, not for the other test engines/frameworks that's built on top of it. As for Spek's plugin, I know it's lacking but we are working hard on adding features to it.

**Comment by denghongcai** [comment]
@raniejade how about add these tips to spek official document? I have a very hard day until find this issue.

**Comment by raniejade** [comment]
@denghongcai I'm in the process of improving the docs, just finishing up some coding tasks.

**Comment by raniejade** [comment]
Resolved by ce4362714ce971223e7368aa5b2f29eee6fa59dc.
